File: potter-frontend/src/types/recommendation.ts

export interface RecommendationOption {
  type: "recommendation";
  id: string;
  title: string;
  summary: string;
  steps: string[];
  frequency: string | null;
  duration: string | null;
  materials: string[];
  caution: string | null;
  expected_result: string;
  recommendation_score: number;
}

export interface AIRecommendationResponse {
  type: "recommendation_options";
  options: RecommendationOption[];
}

File: backend/app/schemas/recommendation.py
from typing import Literal

from pydantic import BaseModel, Field


class RecommendationOption(BaseModel):
    type: Literal["recommendation"] = "recommendation"
    id: str
    title: str
    summary: str
    steps: list[str] = Field(min_length=1)
    frequency: str | None = None
    duration: str | None = None
    materials: list[str] = []
    caution: str | None = None
    expected_result: str
    recommendation_score: int = Field(ge=1, le=5)


class AIRecommendationResponse(BaseModel):
    type: Literal["recommendation_options"] = "recommendation_options"
    options: list[RecommendationOption] = Field(min_length=1)
